### ISSUE TITLE
openssh authorizedKeysFiles breaks nixos-anywhere

### ISSUE THREAD

**Issue by sedlund** [nix-community/srvos#337]
nixos-anywhere generates a temporary ssh key for deployment.  it puts it into $HOME/.ssh which https://github.com/nix-community/srvos/blob/cd802e2933c567ea91de48dbe8968f41a5d9a642/nixos/common/openssh.nix#L30

won't allow for security.

How can I use the nixpkgs default in my flake?

**Comment by phaer** [comment]
Just to clarify: You want to re-install NixSS by running nixos-anywhere against a machine which is already running NixOS with srvos included?

> How can I use the nixpkgs default in my flake?
```
    services.openssh.authorizedKeysFiles = lib.mkOverride 40
      [ "%h/.ssh/authorized_keys" "/etc/ssh/authorized_keys.d/%u" ];
```
(`mkForce` equals `mkOverride 50`)

**Comment by sedlund** [comment]
Yes thats right.  Using disko to re-layout drives / lvm / luks / filesystems.  Thanks for the tip!

**Comment by Mic92** [comment]
We do not set this option in `nixos-images`, so it's unclear to me why this is causing problems with nixos-anywhere?

**Comment by sedlund** [comment]
The problem comes when a system that is built with default SrvOS is initially connected to with `nixos-anywhere` cli  (to be repartitioned / reloaded etc)

SrvOS generates in `/etc/ssh/sshd_config` this line
`AuthorizedKeysFile /etc/ssh/authorized_keys.d/%u`

nixos-anyhwere will generate temp ssh keys, [ssh-copy-id](https://github.com/nix-community/nixos-anywhere/blob/daf19effbafba2cfeac4c41e17dccdd07ca86cb3/src/nixos-anywhere.sh#L318-L325)'s them to the current NixOS `$HOME/.ssh`, then attempts to [import_facts](https://github.com/nix-community/nixos-anywhere/blob/daf19effbafba2cfeac4c41e17dccdd07ca86cb3/src/nixos-anywhere.sh#L346) using the [ssh_](https://github.com/nix-community/nixos-anywhere/blob/daf19effbafba2cfeac4c41e17dccdd07ca86cb3/src/nixos-anywhere.sh#L207-L209) function which specifies these temp keys  which are not allowed.

It has been a long while since this occurred, so I believe  this is what happened.

**Comment by Mic92** [comment]
Ah. Now I get it.
